llvm/test/Transforms/FunctionAttrs/nocapture.ll

```diff
@@ -371,7 +371,6 @@ define void @nc4(ptr %p) {
 ; ATTRIBUTOR: Function Attrs: nosync nounwind memory(read)
 ; ATTRIBUTOR-LABEL: define void @nc4
 ; ATTRIBUTOR-SAME: (ptr nocapture readonly [[P:%.*]]) #[[ATTR7:[0-9]+]] {
-; ATTRIBUTOR-NEXT:    call void @external(ptr nocapture readonly [[P]]) #[[ATTR4]]
```

Comment:
This doesn't look right. Why was the function call removed?

Reply:
Because of the added dead part

Reply:
I mean, that makes the test meaningless.

Reply:
Any idea why AAIsDead is needed?

Reply:
But it doesn't need to know that, it can just assume no callsites are dead. The behavior without it makes no sense

Reply:
No, AA works in a way that, it assumes all uses are dead at the beginning and then makes uses alive from iteration to iteration.

Reply:
For some reason if I make it not dead, it also breaks the inference of nocapture: https://godbolt.org/z/oWs5ana4c

Reply:
I guess it would need some kind of no-argument-return thing

Reply:
I don't see a way to make the test not-meaningless. If the function doesn't write memory, it has to have a return value to be meaningful. If it has a return value, it's not eligible for nocapture